src/tup/Tupfile

```diff
@@ -1,3 +1,10 @@
 include_rules
 : foreach *.c | ../luabuiltin/luabuiltin.h |> !cc |>
-: foreach *.c | ../luabuiltin/luabuiltin.h |> !mingwcc |>
+ifneq (@(TUP_PLATFORM),win32)
+: |> version=`git describe` ; echo "const char *tup_version(void) {return \"$version\";}" >version.c |> version.c
```

Comment:
Thankfully you have solved the issue with the bizarre output file with bdae36034b39e02f39931fe9f5ff2199cab4cb8c. It might have had something to do with the specific version of git I was using. Still, the rule uses syntax specific to Unix-style shells and doesn't work here (cmd.exe complains about invalid syntax), so an alternative for Windows seems necessary. I'll provide one soon.